### PR TITLE
Add page about the high-level architecture of the service

### PR DESCRIPTION
**Pull request by TheDoubleK**
This has the architecture diagram and a brief description of how I think it all hangs together.

Suggestions and amendments welcome!

### CODE REVIEW

File: source/architecture-partials/architecture.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## The Publisher application
+
+The Publisher is a [Python](https://www.python.org/) web application based on the 
+[Flask web framework](http://flask.pocoo.org/) backed by a [Postgres](https://www.postgresql.org/) database.
+
+The Publisher app is hosted on Heroku and released through a three-stage pipeline:
+
+1. **Review app** When a pull request is opened on the Publisher in GitHub the tests are run and a review app is created
+   in Heroku.
+   Review apps get a database copied from the Staging database, but with a reduced number of topics and measure pages
```
Comment (samuelhwilliams) [inline source/architecture-partials/architecture.md]: Same number of topics and subtopics, but no more than two measures per subtopic.

File: source/architecture-partials/architecture.md [hunk 2]
```diff
@@ -93,3 +93,8 @@ by Sentry, which allows issue tracking and captures a detailed stack trace and d
 
 We also have [Sqreen](https://www.sqreen.com/) installed inside Publisher, which monitors activity, blocks some 
 suspicious requests and reports several key site metrics.
+
+## Architecture decision records (ADRs)
```
Comment (samuelhwilliams) [inline source/architecture-partials/architecture.md:98]: 👍

File: source/architecture-partials/architecture.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+   of user accounts set up for each different role.
+
+3. **Production** Promoting code to production is a manual step: use the "Promote to production..." button in the 
+   Heroku pipeline.  This will immediately release the master branch code to production, with no further test run.
+
+## Static site build
+
+The Publisher app allows users to create, update, edit and preview pages for the Ethnicity Facts and Figures site. 
+For these changes to be reflected in the pages of the live site they need to be built and pushed to the production S3
+bucket. A build of the entire site is requested by Publisher whenever a page is marked as published (or, in rare cases,
```
Comment (samuelhwilliams) [inline source/architecture-partials/architecture.md:43]: Also when new code is released, I think?

File: source/architecture-partials/architecture.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+runs nightly to keep these views up to date. (The views are also refreshed right before every static build of the site, 
+so that dashboards published on the live site are always up-to-date.)
+
+We keep the data in our Staging database up-to-date by running the `pull_prod_data` command nightly.  This copies data
+from the production database into the staging database, removing all user email addresses and passwords and setting up
+some known user accounts that can be used for testing in Staging and on review apps.
+
+## Heroku add-on services
+### Email
+We use Mailgun for sending emails from Publisher.  The only emails we send from the app are for users to confirm new
```
Comment (samuelhwilliams) [inline source/architecture-partials/architecture.md:77]: Maybe worth talking about how we've got email set up?

Should we also talk about WorkMail?

Comment (TheDoubleK) [reply source/architecture-partials/architecture.md:77]: I wasn't going to mention WorkMail, but could?

File: source/architecture-partials/architecture.md (hunk 3 above)
Comment (TheDoubleK) [reply source/architecture-partials/architecture.md:43]: I 100% meant to put that in.

File: source/architecture-partials/architecture.md (hunk 4 above)
Comment (samuelhwilliams) [reply source/architecture-partials/architecture.md:77]: Probably worth mentioning, as otherwise easy for that to get totally forgotten.

Comment (TheDoubleK) [reply source/architecture-partials/architecture.md:77]: Does it live here though?  It feels like a separate thing from the architecture.

Comment (TheDoubleK) [reply source/architecture-partials/architecture.md:77]: And what kind of thing would you add about how email is set up?

Comment (TheDoubleK) [reply source/architecture-partials/architecture.md:77]: I've added a note about WorkMail and Secrets Manager and updated the diagram.  See what you reckon.

**Review by samuelhwilliams — APPROVED** [review]
Very nice 👏

**Pull request merged**